### PR TITLE
Backporting 2 RTS Metrics fixes

### PR DESCRIPTION
**Pull request by mathewc**
Merging https://github.com/Azure/azure-functions-host/pull/6913 into V2. Also pulling in another V3 fix in this same area that wasn't merged back to V2 (https://github.com/Azure/azure-functions-host/pull/6539).

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [x] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Scale/TableEntityConverter.cs [hunk 1]
```diff
@@ -91,9 +91,12 @@ public static bool TryGetEntityProperty(JProperty property, out EntityProperty e
                     entityProperty = new EntityProperty(value.ToObject<Guid>());
                     return true;
                 case JTokenType.Integer:
-                    entityProperty = new EntityProperty(value.ToObject<int>());
+                    // to handle both ints and longs, we normalize integer values
+                    // to type long
+                    entityProperty = new EntityProperty(value.ToObject<long>());
```
Comment (soninaren) [inline src/WebJobs.Script.WebHost/Scale/TableEntityConverter.cs:96]: Is this related to the current issue?

**Review by soninaren — APPROVED** [review]

File: src/WebJobs.Script.WebHost/Scale/TableEntityConverter.cs (hunk 1 above)
Comment (mathewc) [reply src/WebJobs.Script.WebHost/Scale/TableEntityConverter.cs:96]: No - please see the PR description above.
